### ISSUE TITLE
Subcommands can be written twice during invocation

### ISSUE THREAD

**Issue by mwcz** [drashland/line#27]
## Summary

A clear and concise summary of the bug.

## Steps To Reproduce The Bug

Using this `cli.ts`

```typescript
import * as Line from "https://deno.land/x/line@v1.0.0/mod.ts";

class MainCommand extends Line.MainCommand {
    public signature = "main";
    public description = "The main command.";

    public subcommands = [
      Subcommand,
    ];

    public handle(): void {
        console.log("Main command.");
    }
}

class Subcommand extends Line.Subcommand {
    public signature = "sub";
    public description = "A subcommand.";

    public handle() {
        console.log("The subcommand.");
    }
}

// Construct your Line.CLI object and plug in your main command

const cli = new Line.CLI({
    name: "A main command with a subcommand.", // Required config
    description: "To reproduce subcommands being written twice.", // Required config
    version: "v1.0.0", // Required config
    command: MainCommand, // Required config
});

// Run your CLI

cli.run();
```

1. `deno run cli.ts` - prints help :heavy_check_mark: 
2. `deno run cli.ts sub` - prints "A subcommand." :heavy_check_mark: 
3. `deno run cli.ts sub foo` - prints "[ERROR] Subcommand 'sub' used incorrectly." :heavy_check_mark: 
4. `deno run cli.ts sub sub` - prints "A subcommand." :warning: :thinking: 

## Expected Behavior

I would have expected a subcommand name can be written only once, and writing `sub sub` would result in "[ERROR] Subcommand 'sub' used incorrectly." 

## Suggested Solution(s)

Unsure, I haven't looked at the code.

## Screenshots

[![asciicast](https://asciinema.org/a/nZqgSwOqolzaGlEHBm31aRT5A.svg)](https://asciinema.org/a/nZqgSwOqolzaGlEHBm31aRT5A)

## Additional Context

none

**Comment by crookse** [comment]
valid, this is definitely a bug

**Comment by dXu23** [comment]
The issue appears to be on lines 30-31. I managed to get the the code snippet cli.ts to work by deleting those lines, but I probably have broken something in the process. 

https://github.com/drashland/line/blob/6accd657213c26d34d05a2251dd3a7e9b9adced5/src/arg_parser.ts#L17-L32

**Comment by dXu23** [comment]
Actually, now that I've thought about this, I'm not too sure this is a bug. While exploring the tests, I noticed that there was a typescript file called cli.ts in the various file_manager directories, which has a subcommand called read, which reads the contents of a file and prints it to standard output. If there is a file called 'read' in the directory, then
```
deno run --allow-read cli.ts read read
```
should not throw out an error for putting a subcommand twice, but it should attempt to read the file named 'read'. So the subcommand actually is only being written once; in the context of @mcwz 's example, line merely interprets the second 'sub' as an argument.

**Comment by dXu23** [comment]
My bad. I think I had the lines 31-32 commented out when I did the test lmao.

**Comment by crookse** [comment]
@dXu23 haha no worries. I think there are a couple of issues:

1. [`arg_parser.ts#extractArgumentsFromDenoArgs:L30-L31`](https://github.com/drashland/line/blob/main/src/arg_parser.ts#L30-L31) is run twice for subcommands.
    1. First run: [`subcommand.ts#run:L61-L62`](https://github.com/drashland/line/blob/main/src/subcommand.ts#L61-L62)
    3. Second run: [`arg_parser.ts#extractArgumentsFromDenoArgs:L30-L31`](https://github.com/drashland/line/blob/main/src/arg_parser.ts#L30-L31)
2. [`main_command.ts#run:L64-L66`](https://github.com/drashland/line/blob/main/src/main_command.ts#L64-L66) doesn't account for the following case: If this main command DOES NOT take in arguments and DOES take in subcommands, then it should error out if the input in `Deno.args` DOES NOT match with one of the subcommands.

I did some quick fixing for bullet 1, but it surfaced bullet 2. Fixing bullet 2 wasn't as straightforward as i thought it would be. i thought i could just add a check for that case, but it breaks tests.

**Comment by crookse** [comment]
@mwcz @dXu23 , just released v1.0.1 which includes fixing this issue. should be good to go :)

**Comment by mwcz** [comment]
Awesome, thank you!
